libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -1013,29 +1028,26 @@ void AP_TECS::update_pitch_throttle(int32_t hgt_dem_cm,
     // Calculate pitch demand
     _update_pitch();
 
-    // Write internal variables to the log_tuning structure. This
-    // structure will be logged in dataflash at 10Hz
-    log_tuning.hgt_dem  = _hgt_dem_adj;
-    log_tuning.hgt      = _height;
-    log_tuning.dhgt_dem = _hgt_rate_dem;
-    log_tuning.dhgt     = _climb_rate;
-    log_tuning.spd_dem  = _TAS_dem_adj;
-    log_tuning.spd      = _integ5_state;
-    log_tuning.dspd     = _vel_dot;
-    log_tuning.ithr     = _integ6_state;
-    log_tuning.iptch    = _integ7_state;
-    log_tuning.thr      = _throttle_dem;
-    log_tuning.ptch     = _pitch_dem;
-    log_tuning.dspd_dem = _TAS_rate_dem;
-    log_tuning.flags    = _flags_byte;
-    log_tuning.time_us  = AP_HAL::micros64();
-}
-
-// log the contents of the log_tuning structure to dataflash
-void AP_TECS::log_data(DataFlash_Class &dataflash, uint8_t msgid)
-{
-    log_tuning.head1 = HEAD_BYTE1;
-    log_tuning.head2 = HEAD_BYTE2;
-    log_tuning.msgid = msgid;
-    dataflash.WriteBlock(&log_tuning, sizeof(log_tuning));
+    // log to DataFlash
+    DataFlash_Class::instance()->Log_Write("TECS", "TimeUS,h,dh,hdem,dhdem,spdem,sp,dsp,ith,iph,th,ph,dspdem,w,f", "QfffffffffffffB",
```

Comment:
Oh, I just noticed that the DataFlash_Class is now a static singleton! That makes things soooo much nicer.